optax/transforms/_freezing_test.py

```diff
@@ -0,0 +1,283 @@
+# Copyright 2019 DeepMind Technologies Limited. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Tests for methods in `optax.transforms._freezing.py`."""
+
+from absl.testing import absltest
+from absl.testing import parameterized
+import chex
+import jax
+import jax.numpy as jnp
+from optax._src import alias
+from optax._src import base
+from optax._src import update
+from optax.transforms import _freezing
+
+
+# Data setup for Freeze shortcuts test
+PARAMS_FLAT = {'a': jnp.zeros(1), 'b': jnp.ones(2)}
+PARAMS_NESTED = {
+    'layer1': [jnp.zeros((2, 3)), {'bias': jnp.ones(3)}],
+    'layer2': {'w': jnp.full((4, 2), 2.0), 'b': jnp.full(4, -1.0)},
+    'misc': (jnp.array(5.0),)
+}
+GRAD_FLAT = {'a': jnp.array([1.0]), 'b': jnp.array([2.0, 3.0])}
+GRAD_NESTED = {
+    'layer1': [jnp.full((2, 3), 1.0), {'bias': jnp.full(3, 2.0)}],
+    'layer2': {'w': jnp.full((4, 2), 3.0), 'b': jnp.full(4, 4.0)},
+    'misc': (jnp.array(1.0),)
+}
+
+
+class FreezeTest(parameterized.TestCase):
+
+  @parameterized.named_parameters([

```

Comment:
Can you add a partial prefix mask as a test case?

The tests have a full prefix mask and a full mask, but we're missing a partial prefix mask like:
```
params = [a, [b, c]]
mask = [False, True]
```